CRM/Contribute/Form/Task/Invoice.php

```diff
@@ -312,7 +312,10 @@ public static function printPDF($contribIDs, &$params, $contactIds) {
           $creditNoteId = $contribution->creditnote_id;
         }
       }
-      $invoiceNumber = CRM_Utils_Array::value('invoice_prefix', $prefixValue) . "" . $contribution->id;
+      $invoiceNumber = $contribution->invoice_number;
+      if (!$invoiceNumber) {
+        $invoiceNumber = CRM_Contribute_BAO_Contribution::storeInvoiceNumber($contribution->id);
```

Comment:
set invoice number for old contribution if invoice_number IS NULL

Reply:
Reading the code, this situation should never happen right?

Reply:
@colemanw this part is for those contributions which has invoice_number set before the fix introduced in ```..._BAO_Contribution::create(...)``` function. Isn't that right?